src/NServiceBus.Core/Serializers/XML/XmlSerializerCache.cs

```diff
@@ -23,8 +23,6 @@ public void InitType(Type t)
 
             if (typeof(XContainer).IsAssignableFrom(t))
             {
-                typesBeingInitialized.Add(t);
```

Comment:
@timbussmann As discussed on the call, once you verify this change and why this line was here in the first place, I think this is good to go.

Reply:
@bording thinking a bit more about this, I think the only potential reason for this statement here is for optimization purposes. It would effectively prevent multiple `IsAssignableFrom` checks for XContainer types. This would actually work with the new code, as we skip property/field initialization of it is already present in the collection (as currently, we will always call InitType for XContainer types, as it won't be added to the list). Interestingly, I don't think that would have worked with the previous code, as the collection isn't checked before initializing type members. I'm not sure this optimization is worth to add this line back again (as I think this hasn't worked before this PR anyway).